File: DS-Algo-OU/Data Structures/Graphs/BFS.py
# take input of the no. of nodes a graph contains
no_of_nodes = int(input("No. of nodes: "))

# array to store all the edges
edges = []

# take input of the edges
while (True):

    # ask for user to continue entering for the edges
    choice = int(input("Do you want to continue: "))

    if (choice != 1):
        break

    # ask user to give input of edge
    edge = input("Enter edge: ")
    edges.append(edge)

# building an adjacency matrix out of the edges given as input


def buildAdjacencyMatrix(edges):

    # build an empty matrix of size n*n
    matrix = []
    for x in range(no_of_nodes):
        matrix.append([0] * no_of_nodes)

    for x in edges:
        temp = x.split('-')
        matrix[int(temp[0]) - 1][int(temp[1]) - 1] = 1

    return matrix


adjacencyMatrix = buildAdjacencyMatrix(edges)

# BFS:
# 1) Start from a node and note down all the non visited vertices into a queue
# 2) mark the choosen node as visited
# 3) now pop out a node from the queue and repeat from the step 1

def BFS_traversal(matrix):

    visited_array = [0] * no_of_nodes

    queue = []

    # 1) Start from the first row of the matrix
    for x in range(len(matrix)):

        # 2) copy all non visited vertices into a queue array
        for y in range(len(matrix[x])):
            

File: DS-Algo-OU/Sreehan/BFS.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// 1) Take input of the no. of nodes a graph has
// 2) Take input of all the edges the graph has
// 3) represent the graph in adjacency matrix
// 4) Create a stack to push all the non visited nodes to the stack and then push and pop accordingly

int *AdjacencyMatrix(int no_of_nodes)
{
    int *matrix = (int *)calloc((no_of_nodes * no_of_nodes), sizeof(int));
    return matrix;
}
void main()
{
    // 1) Take input of the no. of nodes a graph has
    int no_of_nodes;
    printf("Enter the no. of nodes a graph has:\n");
    scanf("%d", &no_of_nodes);
    int *matrix = AdjacencyMatrix(no_of_nodes);

    // 2) Take input of all the edges the graph has
    char edges[10];
    int stopper = 0;
    while (1)
    {
        printf("Enter edges for the graph: ");
        gets(edges);

        char *token = strtok(edges, ' ');
        puts(token);

        printf("Press 1 to stop, Press 0 to contine: ");
        scanf("%d", &stopper);
        if (stopper == 1)
            break;
    }
}